codemagic.yaml

```diff
@@ -4,10 +4,10 @@ definitions:
     max_build_duration: 30
 
   env_versions: &env_versions
-    flutter: 3.22.3
-    xcode: 15.4
+    flutter: 3.27.1
+    xcode: 16.2
     cocoapods: 1.16.2
-    java: 17
+    java: 21
```

Comment:
With JDK21 we get those warning while building the APK:
```
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
3 warnings
```
According to this [thread](https://github.com/flutter/flutter/issues/156111#issuecomment-2391954320) these warnings are harmless in our case.